src/server/pps/server/api_server.go

```diff
@@ -1322,6 +1322,9 @@ func (a *apiServer) GetLogs(request *pps.GetLogsRequest, apiGetLogsServer pps.AP
 		var err error
```

Comment:
I think you might be able to get rid of the `var err error` declaration too? (or, if not, perhaps move it into the `if`/`else` blocks? It seems slightly weird to me that this would always be nil at the next join point in the code but it's not at all a big deal)

Reply:
I kind of like having it here, because we can't use `:=` for the lines where it's set since that will prevent the `pipelineInfo` being passed out of the branches of the `if`. Declaring it inside the blocks would work, but it means declaring it twice which is a bit of a pain.